angular/projects/common/src/app/utils/utils.ts

```diff
@@ -38,6 +40,36 @@ export class Utils {
     return legistrationStrings.join(' ');
   }
 
+  /**
+   * Given an Entity object, return a formatted entity name string.
+   *
+   * @static
+   * @param {Entity} obj
+   * @returns {string} formatted entity string, or empty string if no entity values are set.
+   * @memberof Utils
+   */
+  public static buildEntityString(obj: Entity): string {
+    if (!obj) {
+      return '';
+    }
+
+    if (obj.type === ENTITY_TYPE.Company) {
+      return obj.companyName;
+    }
+
+    if (obj.type === ENTITY_TYPE.IndividualCombined) {
+      return obj.fullName;
+    }
+
+    if (!obj.firstName && !obj.middleName && !obj.lastName) {
+      return '';
+    }
+
+    if (obj.type === ENTITY_TYPE.Individual) {
+      return [[obj.lastName || '-', obj.firstName || '-'].join(', '), obj.middleName || '-'].join(' ');
+    }
+  }
+
```

Comment:
None of this belongs in a common utils class, it belongs to the Entity class itself as a member function.